### DISCUSSION TITLE
SaveImaged Saving NIFTI files does not work correctly

### DISCUSSION THREAD

**Discussion by bxdgit** [Project-MONAI/MONAI#6044] [Q&A] [answered]
**Describe the bug**
I use SaveImaged to save the file, which changes the spacing information of the file。image meta spacing(0.1540, 0.1540, 1.0000)，But the spacing written in the file is (0.1540, 1.0000, 1.0000)

**To Reproduce**
The following code can reproduce the problem
```
import os
import argparse
import numpy as np
from monai.config import KeysCollection
from monai.data import Dataset, DataLoader, decollate_batch
from monai.inferers import sliding_window_inference
from monai.networks.nets import UNet
from monai.transforms import (
    MapTransform,
    Compose,
    Activationsd,
    AsDiscreted,
    Invertd,
    AddChanneld,
    LoadImaged,
    LoadImage,
    NormalizeIntensityd,
    Orientationd,
    Spacingd,
    EnsureChannelFirstd,
    SaveImaged,
    SqueezeDimd,
    ScaleIntensityd,
    CropForegroundd,
    EnsureTyped,
    ToDeviced
)
import torch


class Permute(MapTransform):
    def __init__(self, keys: KeysCollection, allow_missing_keys: bool = False) -> None:
        super().__init__(keys, allow_missing_keys)
    def __call__(self, data):
        d = dict(data)
        # print(d["pred"].meta)
        # print(d["image"].meta)
        # d["pred"].meta = d["image"].meta
        for key in self.keys:
            if key == "pred":
                print(d[key].shape)
                print(type(d[key]))
                d[key] = d[key].permute(1,2,0)
                print(d[key].shape)
                print(d[key].meta["spacing"])
               
        return d

def run_inference(model_file:str, input, output_dir=None,  VAL_AMP=False, print_log=True):
    if output_dir is None and os.path.isdir(input):
        output_dir = input
    # threshold at -1024
    def threshold_at_one(x):
        return x > -1024

    pre_transforms = Compose(
        [
            LoadImaged(keys=["image"]),
            SqueezeDimd(keys=['image'], dim=-1),
            EnsureChannelFirstd(keys=["image"]),
            # CropForegroundd(keys=["image"],source_key="image"),
            CropForegroundd(keys=["image"],source_key="image",select_fn=threshold_at_one,margin=0),
            EnsureTyped(keys=["image"])
        ]
    )

    post_transforms = Compose([
        EnsureTyped(keys="pred"),
        Activationsd(keys="pred", sigmoid=True),
        AsDiscreted(keys="pred", threshold=0.5),
        Permute(keys="pred"),
        Invertd(
            keys="pred",
            transform=pre_transforms,
            orig_keys="image",
            nearest_interp=False,
            to_tensor=True,
        ),

    
        SaveImaged(
            keys="pred",
            output_dir=output_dir,
            output_postfix="label",
            output_dtype=np.uint8,
            resample=False,
            separate_folder=False,
            print_log=print_log,
        ),
    ])

    if os.path.isdir(input):
        images = [
            {
                "image": os.path.join(input, fname),
            }
            for fname in os.listdir(input)
            if fname.endswith(".nii.gz")
        ]
    else:
        images=[{"image":input}]
  
    dataset = Dataset(data=images, transform=pre_transforms)
    data_loader = DataLoader(dataset, batch_size=1, shuffle=False, num_workers=0)

    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

    model = UNet(
            2,
            in_channels=1,
            out_channels=1,
            channels=(16, 32, 64, 128, 256),
            strides=(2, 2, 2, 2),
            num_res_units=2,
        ).to(device)
    # model.load_state_dict(
    #     torch.load(model_file))
    model.eval()
    # define inference method
    def inference(input):

        def _compute(input):
            return sliding_window_inference(
                inputs=input,
                roi_size=(192, 192),
                sw_batch_size=4,
                predictor=model,
            )

        if VAL_AMP:
            with torch.cuda.amp.autocast():
                return _compute(input)
        else:
            return _compute(input)

    with torch.no_grad():
        for data in data_loader:
            image = data["image"].to(device)
            data["pred"] = inference(image)
            # post_transforms(data)
            _ = [post_transforms(i) for i in decollate_batch(data)]


def main():
    run_inference("model file","inputfile", "output",print_log=True)

if __name__ == "__main__":
    main()

```
The input parameter model file can be null. here is my input file
[img949.zip](https://github.com/Project-MONAI/MONAI/files/10801617/img949.zip)

**Expected behavior**
I expect the spacing information of the saved file to be the same as the image meta information.
As you can see from the screenshot below, it changes the spacing information

**Screenshots**
![1677054003694](https://user-images.githubusercontent.com/11461388/220563466-520dfd4d-3e29-4269-bbcc-50355025bb60.jpg)
![1677054175326](https://user-images.githubusercontent.com/11461388/220563500-aac3291f-4b14-405d-8455-14bfecd0b1e1.jpg)

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
================================
Printing MONAI config...
================================
MONAI version: 1.1.0
Numpy version: 1.21.5
Pytorch version: 1.11.0+cu113
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: a2ec3752f54bfc3b40e7952234fbeb5452ed63e3
MONAI __file__: C:\Users\Bxd\AppData\Local\Programs\Python\Python39\lib\site-packages\monai\__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.10
Nibabel version: 3.2.2
scikit-image version: 0.19.2
Pillow version: 9.1.0
Tensorboard version: 2.8.0
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: 0.12.0+cpu
tqdm version: 4.64.0
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: 5.9.0
pandas version: 1.2.4
einops version: 0.5.0
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: 0.4.3

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Windows
Win32 version: ('10', '10.0.19044', 'SP0', 'Multiprocessor Free')
Win32 edition: Professional
Platform: Windows-10-10.0.19044-SP0
Processor: Intel64 Family 6 Model 167 Stepping 1, GenuineIntel
Machine: AMD64
Python version: 3.9.7
Process name: python.exe
Command: ['C:\\Users\\Bxd\\AppData\\Local\\Programs\\Python\\Python39\\python.exe', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='C:\\Program Files\\WindowsApps\\Microsoft.LanguageExperiencePackzh-CN_19041.57.180.0_neutral__8wekyb3d8bbwe\\Windows\\System32\\zh-CN\\KernelBase.dll.mui', fd=-1), popenfile(path='C:\\Program Files\\WindowsApps\\Microsoft.LanguageExperiencePackzh-CN_19041.57.180.0_neutral__8wekyb3d8bbwe\\Windows\\System32\\zh-CN\\kernel32.dll.mui', fd=-1)]
Num physical CPUs: 8
Num logical CPUs: 16
Num usable CPUs: 16
CPU usage (%): [6.2, 4.5, 10.2, 1.9, 5.1, 18.5, 16.6, 51.0, 5.1, 3.2, 4.5, 5.1, 3.2, 4.5, 2.5, 14.6]
CPU freq. (MHz): 3600
Load avg. in last 1, 5, 15 mins (%): [0.0, 0.0, 0.0]
Disk usage (%): 73.2
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 127.8
Available memory (GB): 108.9
Used memory (GB): 18.9

================================
Printing GPU config...
================================
Num GPUs: 1
Has CUDA: True
CUDA version: 11.3
cuDNN enabled: True
cuDNN version: 8200
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_61', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'compute_37']
GPU 0 Name: NVIDIA GeForce RTX 3090 Ti
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 84
GPU 0 Total memory (GB): 24.0
GPU 0 CUDA capability (maj.min): 8.6

```

**Additional context**

**Comment by KumoLiu** [comment]
Hi @bxdgit, seem your `Permute` caused this issue. May I know why you permute the prediction? The `1` is channel dim I think. Could you please try to remove `Permute(keys="pred")` in `post_transforms` and try to save it again? Thanks!

**Comment by bxdgit** [comment]
> Hi @bxdgit, seem your `Permute` caused this issue. May I know why you permute the prediction? The `1` is channel dim I think. Could you please try to remove `Permute(keys="pred")` in `post_transforms` and try to save it again? Thanks!

Glad to get your reply，I remove `Permute(keys="pred")` in `post_transforms, but the result is still the same

**Comment by KumoLiu** [comment, marked as answer]
Hi @bxdgit, oh I got it. An additional issue is that `SqueezeDimd(keys=['image'], dim=-1),` in the `pre_transforms`, it changes your input spacing. You could remove this line and update `EnsureChannelFirstd` with the specified channel dim.
```
EnsureChannelFirstd(keys=["image"], channel_dim=-1),
```

Hope it can help you, thanks! Just move it to the discussion.

**Reply by bxdgit** [reply]
hi @KumoLiu ，Thank you, the problem is solved！
